### PR TITLE
Minor change to UML diagram

### PR DESCRIPTION
**Pull request by DominicFuMJ**
I made some minor changes to the UML diagram, where I made the Interest to be a subtype of the Interest. I also made the interest to be an inheritance of the Tag class.

### CODE REVIEW

File: docs/diagrams/BetterModelClassDiagram.puml [hunk 1]
```diff
@@ -14,6 +14,7 @@ Tag <|-- Interest
 UniquePersonList -right-> Person
 
 Person -up-> "*" Tag
+Tag -> "*" Interest
```
Comment (garywongkai) [inline docs/diagrams/BetterModelClassDiagram.puml:17]: This should be either navigability or 1-to-1 relationship though, its not that tag has many none/many interest

File: docs/diagrams/ModelClassDiagram.puml [hunk 2]
```diff
@@ -45,7 +45,7 @@ Person *--> Email
 Person *--> Address
 Person *--> "*" Tag
 Person *--> "*" Schedule
-Tag --> Interest
+Tag <|--  Interest
```
Comment (garywongkai) [inline docs/diagrams/ModelClassDiagram.puml:48]: This here is correct

File: docs/diagrams/ModelClassDiagram.puml [hunk 3]
```diff
@@ -54,7 +54,6 @@ Name -[hidden]right-> Phone
 Phone -[hidden]right-> Address
 Address -[hidden]right-> Email
 Email -[hidden]right-> Tag
-Tag -[hidden]right-> Interest
```
Comment (garywongkai) [inline docs/diagrams/ModelClassDiagram.puml:57]: Yup looks good here

Comment (AwJoey) [reply docs/diagrams/ModelClassDiagram.puml:57]: LGTM!

**Review by AwJoey — APPROVED** [review]

**Review by AwJoey — APPROVED** [review]

File: docs/diagrams/ModelClassDiagram.puml (hunk 3 above)
Comment (DominicFuMJ) [reply docs/diagrams/ModelClassDiagram.puml:57]: Looks good.

File: docs/diagrams/ModelClassDiagram.puml (hunk 2 above)
Comment (DominicFuMJ) [reply docs/diagrams/ModelClassDiagram.puml:48]: Looks good.

**Pull request merged**